### PR TITLE
beekeeper-studio: init at 1.10.2

### PR DESCRIPTION
**Pull request by zoedsoupe**
###### Motivation for this change
Add the open source SQL editor and database manager `beekeeper-studio`! It's agnostic of
which database you use so you can connect to `MySQL`, `MariaDB`, `PostgreSQL` and many others
in a cool GUI app.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+  meta = with lib; {
+    homepage = "https://www.beekeeperstudio.io/";
+    description = "Open Source SQL Editor and Database Manager";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ Mdsp9070 ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:92]:
```suggestion
  };
}
```

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  installPhase = ''
+    mkdir -p $out/share/beekeeper-studio $out/lib $out/bin
+
+    mv usr/share/* $out/share/
+    mv opt/Beekeeper\ Studio/* $out/share/beekeeper-studio/
+    mv $out/share/beekeeper-studio/*.so $out/lib/
+
+    ln -s $out/share/beekeeper-studio/beekeeper-studio-bin $out/bin/beekeeper-studio
+    sed -i 's|\/opt\/Beekeeper Studio|'$out'/bin|g' $out/share/applications/beekeeper-studio.desktop
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:77]: Please use substituteInPlace.

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  version = "1.10.2";
+
+  src = fetchurl {
+    url =
+      "https://github.com/beekeeper-studio/beekeeper-studio/releases/download/v${version}/beekeeper-studio_${version}_amd64.deb";
+    sha256 = "5ada21ded5afd15a9fde1d02a0fe150bd4bba4b3d5cb0cbc533fe4ea1049c720";
+  };
+
+  nativeBuildInputs =
+    [ autoPatchelfHook dpkg makeWrapper gobject-introspection wrapGAppsHook ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:28]:
```suggestion
  nativeBuildInputs = [ autoPatchelfHook dpkg makeWrapper gobject-introspection wrapGAppsHook ];
```

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    openssl
+    stdenv.cc.cc
+  ];
+in stdenv.mkDerivation rec {
+  pname = "beekeeper-studio";
+  version = "1.10.2";
+
+  src = fetchurl {
+    url =
+      "https://github.com/beekeeper-studio/beekeeper-studio/releases/download/v${version}/beekeeper-studio_${version}_amd64.deb";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:23]:
```suggestion
    url = "https://github.com/beekeeper-studio/beekeeper-studio/releases/download/v${version}/beekeeper-studio_${version}_amd64.deb";
```

Can we build from source instead?

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+let
+  runtimeLibs = lib.makeLibraryPath [
+    curl
+    glibc
+    libudev0-shim
+    nghttp2
+    openssl
+    stdenv.cc.cc
+  ];
+in stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:17]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    mv usr/share/* $out/share/
+    mv opt/Beekeeper\ Studio/* $out/share/beekeeper-studio/
+    mv $out/share/beekeeper-studio/*.so $out/lib/
+
+    ln -s $out/share/beekeeper-studio/beekeeper-studio-bin $out/bin/beekeeper-studio
+    sed -i 's|\/opt\/Beekeeper Studio|'$out'/bin|g' $out/share/applications/beekeeper-studio.desktop
+  '';
+
+  preFixup = ''
+    wrapProgram "$out/bin/beekeeper-studio" --prefix LD_LIBRARY_PATH : ${runtimeLibs}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:81]:
```suggestion
    wrapProgram "$out/bin/beekeeper-studio" --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
    curl
    glibc
    libudev0-shim
    nghttp2
    openssl
    stdenv.cc.cc
  ]}
```

File: pkgs/development/tools/database/beekeeper-studio/default.nix [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://www.beekeeperstudio.io/";
+    description = "Open Source SQL Editor and Database Manager";
+    license = licenses.mit;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ Mdsp9070 ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/database/beekeeper-studio/default.nix:89]: Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add Mdsp9070``.

**Comment by lukegb** [comment]
Thanks for the contribution! :) PTAL at the outstanding review comments.

**Comment by matdsoupe** [comment]
https://github.com/NixOS/nixpkgs/pull/140700 solves this package!

**Pull request closed without merging**